src/data/ACTIONS/NIN.js

```diff
@@ -31,122 +31,199 @@ export default {
 		},
 	},
 
-	// -----
-	// Player OGCDs
-	// -----
-
 	TEN: {
 		id: 2259,
 		name: 'Ten',
 		icon: 'https://xivapi.com/i/002000/002901.png',
-		onGcd: false,
-		cooldown: 0.5,
+		onGcd: true,
+		cooldown: 20,
+		gcdRecast: 0.5,
 	},
 
 	CHI: {
-		id: 2261,
+		id: 18806,
 		name: 'Chi',
 		icon: 'https://xivapi.com/i/002000/002902.png',
-		onGcd: false,
-		cooldown: 0.5,
+		onGcd: true,
+		cooldown: 20,
+		gcdRecast: 0.5,
 	},
 
 	JIN: {
-		id: 2263,
+		id: 18807,
 		name: 'Jin',
 		icon: 'https://xivapi.com/i/002000/002903.png',
-		onGcd: false,
-		cooldown: 0.5,
+		onGcd: true,
+		cooldown: 20,
+		gcdRecast: 0.5,
 	},
 
 	NINJUTSU: {
 		id: 2260,
 		name: 'Ninjutsu',
 		icon: 'https://xivapi.com/i/002000/002904.png',
-		onGcd: false,
+		onGcd: true,
 		cooldown: 20,
+		gcdRecast: 1.5,
 	},
 
 	FUMA_SHURIKEN: {
 		id: 2265,
 		name: 'Fuma Shuriken',
 		icon: 'https://xivapi.com/i/002000/002907.png',
-		onGcd: false,
+		onGcd: true,
 		cooldown: 20,
+		gcdRecast: 1.5,
+	},
+
+	FUMA_SHURIKEN_TCJ: {
+		id: 18873,
+		name: 'Fuma Shuriken',
+		icon: 'https://xivapi.com/i/002000/002907.png',
+		onGcd: true,
+		cooldown: 20,
+		gcdRecast: 1.5,
```

Comment:
TCJ step 1 and 2 actually have 1.0 gcdRecast (surprised me when i checked it).  Also, like ninjutsu, technically no cooldown.

Reply:
Also valid; I'll correct the data here as well.